pkg/controller/transfer/pvc.go

```diff
@@ -0,0 +1,171 @@
+	pvcExists, err := h.reconciler.getSourceResource(ot, pvc)
+	if err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	if !pvcExists {
+		// will reconcile again when pvc is created/updated
+		if err := h.reconciler.setAndUpdateCompleteCondition(ot, corev1.ConditionFalse, "No source", ""); err != nil {
+			return 0, err
+		}
+
+		return 0, nil
+	}
+
+	if pvc.Spec.VolumeName == "" || pvc.Status.Phase != corev1.ClaimBound {
+		if err := h.reconciler.setAndUpdateCompleteCondition(ot, corev1.ConditionFalse, "PVC not bound", ""); err != nil {
+			return 0, err
+		}
+
+		return 0, nil
+	}
+
+	pods, err := cdicontroller.GetPodsUsingPVCs(h.reconciler.Client, pvc.Namespace, sets.NewString(pvc.Name), false)
+	if err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	if len(pods) > 0 {
+		if err := h.reconciler.setAndUpdateCompleteCondition(ot, corev1.ConditionFalse, "Pods using PVC", ""); err != nil {
+			return 0, err
+		}
+
+		return defaultRequeue, nil
+	}
+
+	pvc2 := pvc.DeepCopy()
+	pvc2.Status = corev1.PersistentVolumeClaimStatus{}
+	data := map[string]string{
+		"pvName": pvc.Spec.VolumeName,
+	}
+
+	return 0, h.reconciler.pendingHelper(ot, pvc2, data)
+}
+
+func (h *pvcTransferHandler) ReconcileRunning(ot *cdiv1.ObjectTransfer) (time.Duration, error) {
+	pvName, ok := ot.Status.Data["pvName"]
+	if !ok {
+		if err := h.reconciler.setAndUpdateCompleteCondition(ot, corev1.ConditionFalse, "PV name missing", ""); err != nil {
+			return 0, err
+		}
+
+		return 0, nil
+	}
+
+	pv := &corev1.PersistentVolume{}
+	if err := h.reconciler.Client.Get(context.TODO(), types.NamespacedName{Name: pvName}, pv); err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	reclaim, ok := ot.Status.Data["pvReclaim"]
+	if !ok {
+		ot.Status.Data["pvReclaim"] = string(pv.Spec.PersistentVolumeReclaimPolicy)
+		if err := h.reconciler.setCompleteConditionRunning(ot); err != nil {
+			return 0, err
+		}
+
+		return 0, nil
+	}
+
+	source := &corev1.PersistentVolumeClaim{}
+	sourceExists, err := h.reconciler.getSourceResource(ot, source)
+	if err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	target := &corev1.PersistentVolumeClaim{}
+	targetExists, err := h.reconciler.getTargetResource(ot, target)
+	if err != nil {
+		return 0, h.reconciler.setCompleteConditionError(ot, err)
+	}
+
+	if sourceExists {
+		if pv.Spec.PersistentVolumeReclaimPolicy != corev1.PersistentVolumeReclaimRetain {
+			pv.Spec.PersistentVolumeReclaimPolicy = corev1.PersistentVolumeReclaimRetain
+			if err := h.reconciler.updateResource(ot, pv); err != nil {
+				return 0, h.reconciler.setCompleteConditionError(ot, err)
+			}
+
+			return 0, h.reconciler.setCompleteConditionRunning(ot)
+		}
+
+		if source.DeletionTimestamp == nil {
+			if err := h.reconciler.Client.Delete(context.TODO(), source); err != nil {
+				return 0, h.reconciler.setCompleteConditionError(ot, err)
```

Comment:
Do you need to rollback the PVC reclaim policy here?

Reply:
Nevermind, Maybe I'm just confused about eventual consistency.  This loop will continue trying each time until it gets all the way through.